docs/build/guides/dapps/soroban-contract-init-template.mdx

```diff
@@ -1,442 +1,282 @@
-To install the dependencies and build the typescript bindings for the package, navigate to the `packages` directory, then enter `hello_world` package directory and run the command below.
+Let's try it. First, make sure you're running a local Stellar network. Start Docker Desktop or whatever alternative you prefer, then:
 
 ```bash
-npm i
-npm run build
+stellar network container start local
 ```
 
-After running the command above, there should now be a `node modules` and a `dist` directory containing `index.js` and `index.ts` files. These will serve as an entry point to the package when it is eventually imported.
+Now we can run `contract init`. Let's include one extra contract. Then we'll run the `dev` script (which, remember, will run the `initialize.js` script first):
 
-### Using the hello world package in React
+```bash
+stellar contract init . --with-example increment
+npm run dev
+```
 
-First we import the following from the package:
+After you do this, you may notice lots of new changes in the project. All that Rust, Cargo, and Soroban stuff!
 
-1. A `networks` object.
-2. A `Client` class.
+```bash
+$ git status --short
 
-```javascript
-import { Client, networks } from "../packages/hello_world";
+?? .soroban/
+?? Cargo.lock
+?? Cargo.toml
+?? contracts/
+?? target/
 ```
 
-Notice how we point to the directory of the package when importing th ebindings by making reference to the packages folder.
+Q: Should you commit this stuff?<br />A: **_NO!_**
 
-Now we instantiate a the `Client` class and pass an object containing the RPC url and a copy of the `networks.testnet` object by using the spread operator. This technically means that the `Client` class takes in an object of the network configuration along with the contract ID but we make it easier by using the `networks` object.
+Q: Should you _gitignore_ this stuff?<br />A: **_NO!_**
 
-```javascript
-const helloWorldContract = new Client({
-  ...networks.testnet,
-  rpcUrl: "https://soroban-testnet.stellar.org:443",
-});
-```
+The `.soroban` and `target` folders [are gitignored by the core Rust/Cargo template](https://github.com/stellar/stellar-cli/blob/main/cmd/soroban-cli/src/utils/contract-init-template/.gitignore). In projects that specify yours as their `--frontend-template`, that `.gitignore` will be merged with yours. No need to include those lines twice.
 
-By printing the `helloWorldContract` object, you'll find the hello method which coincides with the hello function defined in the smart contract. Calling this method will invoke its equivalent in the smart contract.
+And projects that use yours will want to commit the `Cargo.*` files and `contracts`, but your _template_ should not include them. That can cause problems with project initialization later. Nor should it gitignore them, because your `.gitignore` will be part of projects started with your template! It would cause problems and confusion, if people didn't realize that some of their most important files were gitignored!
```

Comment:
I think we should allow people to add these files to their templates, and just skip importing them into the created project:

- https://github.com/stellar/stellar-cli/issues/1578

Once we implement that, I'll circle back and update this guide.